### ISSUE TITLE
Missing JaxRS beans in the Jandex index

### ISSUE THREAD

**Issue by loicmathieu** [quarkusio/quarkus#2667]
**Describe the bug**
When returning a `javax.ws.rs.core.Response` in a RestClient, I have the following WARNING in the log, there seems to be some misisng beans in the Jandex index .

```
2019-05-31 09:56:58,610 WARN  [io.qua.dep.ste.ReflectiveHierarchyStep] (build-2) Unable to properly register the hierarchy of the following classes for reflection as they are not in the Jandex index:
	- javax.ws.rs.client.ClientResponseContext
	- javax.ws.rs.core.EntityTag
	- javax.ws.rs.core.Link
	- javax.ws.rs.core.MediaType
	- javax.ws.rs.core.NewCookie
	- javax.ws.rs.core.Response
	- javax.ws.rs.core.Response$StatusType
Consider adding them to the index either by creating a Jandex index for your dependency via the Maven plugin, an empty META-INF/beans.xml or quarkus.index-dependency properties.");.
```
A returning a Response object is a very common thing (when using POST to create an entity for exemple), these should be included in the Jandex index by default. I check the code in SmallRyeRestClientProcessor::processInterfaces and the return types seems to be registered so I'm not sure it's a real issue a just a log issue. And as it works maybe it's just a log issue ;)

**Expected behavior**
No warning in the log file.

**To Reproduce**
1. Create an application following the Rest client tutorial : https://quarkus.io/guides/rest-client-guide
2. Add a method using a remote POST operation and returning a Response object to the CountriesService interface.

Using these simple steps from the tutorial, we also seen a lot a warning from Resteasy saying that some providers are already registred and so the second registration will be ignored, they seems to be ignorable but it's still ugly to have them in the log.

**Environment (please complete the following information):**
 - Quarkus version or git rev: 0.15.0

**Comment by gsmet** [comment]
Ah, yes, we probably have the ignore rules only in the RESTEasy extension.

**Comment by loicmathieu** [comment]
OK, I think I see how to fix this. Is it correct that the way to fix it is the same as what has been done in ResteasyServerCommonProcessor with TYPES_IGNORED_FOR_REFLECTION ?
In this case I can provide a PR for this, I'll be very happy to od it ;)

**Comment by gsmet** [comment]
Yes, exactly. We should probably move the list of ignored types to `ResteasyCommonProcessor` in a public constant so that we can share it.

**Comment by loicmathieu** [comment]
This will create a dependancy between smallrye-rest-client and resteasy-common, if someone want to make a rest-client from a command line application they will have no need to resteasy right?

**Comment by loicmathieu** [comment]
Just saw that resteasy-client already have a dependency to resteasy-common ;)
I will work on it and provide a PR soon.

**Comment by gsmet** [comment]
Yes, `resteasy-common` was added exactly for this purpose.

**Comment by geoand** [comment]
I assume this was fixed by #2667?

**Comment by loicmathieu** [comment]
Yes, the PR is merged but it didn't close the issue automatically. As I can close it, I'll do it myself :)
